alert.go

```diff
@@ -577,7 +590,29 @@ func (a *AlertNode) handleAlert(ad *AlertData) {
 	}
 }
 
-func (a *AlertNode) determineLevel(now time.Time, fields models.Fields, tags map[string]string) AlertLevel {
+func (a *AlertNode) determineLevel(now time.Time, fields models.Fields, tags map[string]string, currentLevel AlertLevel) AlertLevel {
+	// Evaluate reset expression of current level if it exists
+	if cse := a.levelResets[currentLevel]; cse != nil {
+		if pass, err := EvalPredicate(cse, a.lrScopePools[currentLevel], now, fields, tags); err != nil {
+			a.logger.Printf("E! error evaluating reset expression for current level %v: %s", currentLevel, err)
+		} else if !pass {
+			for l := len(a.levels) - 1; l >= 0; l-- {
+				se := a.levels[l]
+				if se == nil {
+					continue
+				}
+				if pass, err := EvalPredicate(se, a.scopePools[l], now, fields, tags); err != nil {
+					a.logger.Printf("E! error evaluating expression for level %v: %s", AlertLevel(l), err)
+					continue
+				} else if pass {
+					if currentLevel < AlertLevel(l) {
+						return AlertLevel(l)
+					}
+				}
+			}
```

Comment:
@minhdanh Now that I see how this is working I agree there is a simpler way to perform this logic.

_A reset expressions only prevents an alert from being lowered in severity_

This mean to simplify we can check just the higher level expressions first and only eval the reset expressions if the new level is not higher.

Psudeo code:

```go
for l = CRITICAL; l > currentLevel; l--{
   eval normal expressions, return the first level that evals to true
}

if resetExpression exists {
 if  !resetExpression {
    return currentLevel
 } else {
   for l = currentLevel - 1; l > OK; l-- {
     eval each lower expression and return the first that is true.
  }
} 
```

Since there are two loops that loops over a subset of levels and return the first that matches you could create a function for that logic. something like `findFirstMatch(start , stop AlertLevel,  all other needed data to eval...) (AlertLevel, error)` 

>NOTE: In my pseudo code I am using the `AlertLevel` type instead of `int` I think using the actual `AlertLevel` type will be more clear instead of `len(a.levels)- 1` since we are now comparing against `currentLevel`

Thanks for taking the time to get this right.